src/if_statements.py

```diff
@@ -762,29 +730,34 @@ def build_body(context: Context, function_info: FunctionInfo, options: Options)
                 if has_common and target == most_common and index != 0:
                     continue
                 context.case_nodes[target].append((switch_index, index))
-        elif isinstance(node, ConditionalNode) and node.is_loop():
-            context.loop_nodes.add(node.conditional_edge)
-        elif isinstance(node, BasicNode) and node.is_loop():
-            context.loop_nodes.add(node.successor)
 
     if options.debug:
         print("Here's the whole function!\n")
+
     body: Body
-    if options.ifs:
-        body = build_flowgraph_between(context, start_node, return_node, 1)
+    if options.ifs and is_reducible:
```

Comment:
There are 15 examples across the oot/mm/papermario corpuses which are non-reducible, they seem to mostly be infinite loops (e.g. `Idle_ThreadEntry`, `crash_monitor_thread`) -- and there's a known path for making those graphs reducible in the future.

The goal of this PR is to make it so that `build_flowgraph_between` and all of the subsequent analysis can assume that the flowgraph is properly reducible. If that doesn't hold, it falls back to the "dumb" --gotos-only mode. 

It's theoretically possible to do the interval/structural analysis "around" the irreducible sections - but we'd need to be much more careful about the calculation & interpretation of `dominators`, `immediate_postdominators`, etc. in a way that I don't think is worth the complexity?

Reply:
I agree it's not worth the complexity, I'm just wondering if it may work sometimes by accident anyway and it could be worth providing that option to the user. (The definition of a reducible cfg seemed pretty complex to me, with three distinct properties, and I was thinking maybe we only rely on e.g. two of them. It may be a naive question because I don't have intuition about reducible cfg's...)

Reply:
I'm going to try to explain it better in the comments, but I think the big picture I was going for is that if the flow graph is reducible, now all of the `build_flowgraph_between`, `build_conditional_subgraph`, `build_switch_between` is pretty straightforward interval analysis with well-defined entry & exit nodes. 

I was able to simplify the logic in these functions because there are fewer edge cases -- but at the cost of not supporting irreducible graphs. The path forward for irreducible graphs is to make them reducible, rather than adding special cases to `build_*`.